quilt_repository_format.md

```diff
@@ -0,0 +1,137 @@
+# Quilt CLI Local Storage
+
+## Motivation
+The primary motivation for unifying the local package store is that any Python process will be able to access any locally installed Quilt package. A common point of friction for Quilt users has been that packages built on the command line aren't visible inside Jupyter notebooks if the Jupyter server is running in a different directory. Unifiying the local storage by adopting the Docker storage model solves that problem and also enables useful future features like comparing (diff) package versions and tracking revision history locally without pushing to a registry.
+
+## Directory Structure
+This change proposes changing from a hierarchy of separate local package stores(quilt_packages) like npm, to a single store like Docker.
+
+```bash
+BASE_DIR/objs/
+```
+Stores binary data objects identified by hash. These objects include compressed raw files and Parquet files. Object hashes are verified and objects are stored only once (deduplication).
+
+```bash
+BASE_DIR/pkgs/
+```
+
+Stores package manifest files (JSON format) for all package instances. Package manifests are identified by the hash of the package.
+
+```bash
+BASE_DIR/contents.json
```

Comment:
We should consider using something like a Sqlite database - or at the very least, lock the file. Otherwise, it'll be too easy to corrupt the data by running multiple quilt processes at the same time.

Reply:
Good point. I'd rather not rely on sqlite and would prefer to keep it both human and machine readable if possible. But, we'll definitely need to have a plan for synchronization.